lemminx-liberty/src/test/java/io/openliberty/LibertyDiagnosticTest.java

```diff
@@ -750,4 +750,115 @@ public void testUniquenessForNameChangedFeatureDiagnostic() throws BadLocationEx
         XMLAssert.testDiagnosticsFor(serverXML, null, null, serverXMLURI,
                 invalid);
     }
+
+    @Test
+    public void testInvalidPlatformDiagnosticWithCodeCompletion() throws BadLocationException {
+
+        String serverXML = String.join(newLine, //
+                "<server description=\"Sample Liberty server\">", //
+                "       <featureManager>", //
+                "               <platform>javaEE</platform>", //
+                "       </featureManager>", //
+                "</server>" //
+        );
+        Diagnostic invalid1 = new Diagnostic();
+        invalid1.setRange(r(2, 25, 2, 31));
+        invalid1.setCode(LibertyDiagnosticParticipant.INCORRECT_PLATFORM_CODE);
+        invalid1.setMessage("ERROR: The platform \"javaEE\" does not exist.");
+
+        XMLAssert.testDiagnosticsFor(serverXML, null, null, serverXMLURI,
+                invalid1);
+        //expecting code action to show all javaee platforms ignoring input case
+        List<String> featuresStartWithJavaEE = new ArrayList<>();
+        featuresStartWithJavaEE.add("javaee-6.0");
+        featuresStartWithJavaEE.add("javaee-7.0");
+        featuresStartWithJavaEE.add("javaee-8.0");
+        Collections.sort(featuresStartWithJavaEE);
+
+        List<CodeAction> codeActions = new ArrayList<>();
+        for (String nextFeature : featuresStartWithJavaEE) {
+            TextEdit texted = te(invalid1.getRange().getStart().getLine(), invalid1.getRange().getStart().getCharacter(),
+                    invalid1.getRange().getEnd().getLine(), invalid1.getRange().getEnd().getCharacter(), nextFeature);
+            CodeAction invalidCodeAction = ca(invalid1, texted);
+
+            codeActions.add(invalidCodeAction);
+        }
+
+        XMLAssert.testCodeActionsFor(serverXML, invalid1, codeActions.get(0), codeActions.get(1),
+                codeActions.get(2));
+
+        serverXML = String.join(newLine, //
+                "<server description=\"Sample Liberty server\">", //
+                "       <featureManager>", //
+                "               <feature>batch-1.0</feature>", //
+                "               <platform>ja</platform>", //
```

Comment:
It would be nice to check that no code actions were given for `ja` still.

Reply:
Added back